### DISCUSSION TITLE
Set tenantID on auth()

### DISCUSSION THREAD

**Discussion by rshibli** [invertase/react-native-firebase#4462] [Q&A] [answered]
Is it possible to set tenantID on auth? It seems that the native module supports this but I haven’t been able to get it to work via JavaScript. 

Google Cloud Identity allows you to create tenants and the firebase.js library allows you to pass the tenant id to auth() to authenticate against a tenants users.

**Comment by mikehardy** [comment]
If you can provide a link to the underlying API on both firebase-ios-sdk and firebase-android-sdk plus the link to firebase-js-sdk for an idea of the API shape we should mimic, then likely yes? A PR to implement the same would be the way to guarantee it was done - dev onboarding setup here to allow for developing the API wrapper https://github.com/invertase/react-native-firebase/blob/master/tests/README.md

**Comment by mikehardy** [comment, marked as answer]
Hey @rshibli  ! This is coming soon, there's a PR #5004 courtesy of @xianlinbox - it is nearly in shape to merge (I need to review again, but it is looking good) - you should subscribe to it and look for it in next release notes

**Reply by rshibli** [reply]
This is fantastic news, and very timely!  Many thanks to you and @xianlinbox!

**Reply by mikehardy** [reply]
@rshibli  This is out as of v11.1.0 - please test and if there's anything wrong let us know - we do not have multi-tenant set up in our test project and the auth emulator does not support multi-tenant so I was able to test it a fair bit, but not completely. That usually means there is something wrong, so I am listening intently on this one and would love to hear that people have tested this and after setting tenant id new users have their tenantId property set correctly

**Reply by rshibli** [reply]
Hi, just wanted to let you know that I migrated back to RNFirebase today and am able to successfully authenticate by setting the tenantId.  I'll continue to test and let you know if there's any issue.
